support/xspec-dev/xspec-mx-html-report.xsl

```diff
@@ -0,0 +1,490 @@
+}
+</xsl:text>
+   </xsl:template>
+   
+   <xsl:template mode="theme-css" priority="1" match=".[.='toybox']">
+      <xsl:text xml:space="preserve" expand-text="false">
+.toybox {
+   .label {   background-color: black; color: white }
+   .failing .label { background-color: darkred }
+   .passing .label { background-color: darkgreen }
+   .pending .label { background-color: inherit; color: black }
+   .passing { background-color: honeydew }
+   .pending { background-color: cornsilk }
+   .failing { background-color: mistyrose }
+   .failing .passing { background-color: honeydew }
+   .pending.zero, .failing.zero { background-color: inherit } 
+}
+</xsl:text>
+   </xsl:template>
+      
+   <xsl:template mode="theme-css" match=".">
+      <xsl:message>No template for theme '{ $theme }' - using 'simple'</xsl:message>
+   </xsl:template>
+   
+   <xsl:template name="page-js">
+      <script type="text/javascript">
+<xsl:text xml:space="preserve" expand-text="false">
+const viewSection = eID => {
+     let who = document.getElementById(eID); 
+     let openers = getAncestorDetails(who);
+     openers.forEach(o => { o.open = true; } );
+     who.scrollIntoView( {behavior: 'smooth'} );
+}
+
+const getAncestorDetails = el => {
+    let ancestors = [];
+    while (el) {
+      if (el.localName === 'details') { ancestors.unshift(el); }
+      el = el.parentNode;
+    }
+    return ancestors;
+}
+
+const expandSectionDetails = inSection => {
+    let sectionDetails = [... inSection.getElementsByTagName('details') ];
+    sectionDetails.forEach(d => { d.open = true; } );
+}
+const collapseSectionDetails = inSection => {
+    let sectionDetails = [... inSection.getElementsByTagName('details') ];
+    sectionDetails.forEach(d => { d.open = false; } );
+}
+
+<!-- thanks to https://www.30secondsofcode.org/js/s/unescape-html/ -->
+ 
+const unescapeHTML = str =>
+  str.replace(
+    /&amp;amp;|&amp;lt;|&amp;gt;|&amp;#39;|&amp;quot;/g,
+    tag =>
+      ({
+        '&amp;amp;': '&amp;',
+        '&amp;lt;': '&lt;',
+        '&amp;gt;': '>',
+        '&amp;#39;': "'",
+        '&amp;quot;': '"'
+      }[tag] || tag)
+  );
+
+
+
+const clipboardCopy = async (who) => {
+      let cp = unescapeHTML(who.innerHTML);
+      try { await navigator.clipboard.writeText(cp); }
+      catch (err) { console.error('Failed to copy: ', err); }
+    }
+
+</xsl:text>
+      </script>
+   </xsl:template>
+
+   <xsl:mode name="count-tests" on-no-match="fail"/>
+   
+   <!-- 'Visitor' pattern matches strings to dispatch processing -->
+   <xsl:template mode="count-tests" match=".[.='passing']">
+      <xsl:param name="where" as="element()"/>
+      <xsl:sequence select="count($where/descendant::x:test[@successful='true'])"/>
+   </xsl:template>
+   
+   <xsl:template mode="count-tests" match=".[.='pending']">
+      <xsl:param name="where" as="element()"/>
+      <xsl:sequence select="count($where/descendant::x:test[matches(@pending,'\S')])"/>
```

Comment:
If the original test uses `<x:pending>` without a label, I think the result XML file will have `pending=""`, which doesn't match `\S`. There may be other parts of this XSLT that need some adjustment if you want to support `<x:pending>` without a label.

Reply:
Thanks! Habit learned from writing XSLTs for other kinds of source data ... we will have to return to `x:pending` when coming back to coverage.